AGENTS.md

```diff
@@ -124,6 +124,27 @@ Some examples include:
 
 [What constitutes a 'Good Commit']: https://www.andrewhowden.com/p/anatomy-of-a-good-commit-message
 
+When writing commit messages, it is crucial to provide structured context for AI agents. This allows them to
+understand the history of the project and make better decisions. To do this, include a special block in your
+commit message, formatted as follows:
+
+```
+# --- BEGIN AGENT CONTEXT ---
+# design: |
```

Comment:
This should be readable primarily by humans please. Agents can also read it, but primary target is human.